### PR TITLE
V1 db models

### LINKED ISSUE #7: Create User and all User related models
The purpose of this task is to create `User` model and all other dynamic models related to user.

**TODO** Add schema of the tables

### LINKED ISSUE #8: Create Skill model and all skill related models
**TODO** Add schema of the tables

### PR DESCRIPTION
**Pull request by filipjurcak**
Closes #7 and closes #8 

This CR adds definitions of all models required for v1. All model definitions follow suggestions specified in [this issue](https://github.com/uk-kona/univerzita-kona-api/issues/5) by @Saamoot.

The models are specified in [this jpg](https://liveuniba.sharepoint.com/sites/UniverzitaKona/Zdielane%20dokumenty/Informatika/Webová%20stránka/UniKonaDB-update.jpg).

Some open questions to think about before (not) approving this PR:
* Do we really need relation tables (for example UserGroup model)? The only reason I can think of right now is the metadata about the relations
* If yes, are we okay with running raw SQL to fetch all the relations of the model? When you specify relation between tables in SQLAlchemy directly, you can easily access all relations of the table, all of this would be wasted with these relation tables and we would probably need to write raw SQL in backend to get all the information we need

### CODE REVIEW

File: app/models.py [hunk 1]
```diff
@@ -0,0 +1,153 @@
+from app import db
+from datetime import datetime
+
+
+class User(db.Model):
+    id = db.Column(db.Integer, primary_key=True)
+    name = db.Column(db.String(50), nullable=False)
+    surname = db.Column(db.String(50), nullable=False)
+    email = db.Column(db.String(100), unique=True, nullable=False)
+    password = db.Column(db.String(100), nullable=False)
```
Comment (matusf) [inline app/models.py]: Do we store the passwords? In plain text? :see_no_evil: :scream:

**Review by matusf — COMMENTED** [review]
From the pythonic point of view, LGTM. From the security & privacy view, I'd not store passwords but rather a hash (of password & salt). The `__repr__` methods have private user data (e.g. mail), I'd be cautious about logging it in the future. As for the DB layout & metadata, I'm not gonna comment, we have "experts" on that.

..okay, I will. According to the GDPR, user has a right to have his/her private information deleted. Marking his/her records as deleted (`deleted_at` fields) and keeping the data stored is effectively violation of GDPR.

File: app/models.py (hunk 1 above)
Comment (filipjurcak) [reply app/models.py]: of course not, that will be hash, even though the name doesn't suggest it
so I will change the name to make it clear

**Comment by Saamoot** [comment]
```
According to the GDPR, user has a right to have his/her private information deleted. Marking his/her records as deleted (deleted_at fields) and keeping the data stored is effectively violation of GDPR.
```
Deleted fields are for easier data recovery, if request to remove private information is made data should be removed according to  law.

```
Do we really need relation tables (for example UserGroup model)? The only reason I can think of right now is the metadata about the relations

If yes, are we okay with running raw SQL to fetch all the relations of the model? When you specify relation between tables in SQLAlchemy directly, you can easily access all relations of the table, all of this would be wasted with these relation tables and we would probably need to write raw SQL in backend to get all the information we need
```
Do not write raw sql unless absolutelly necessary. SqlAlchemy support relation tables using:
-  ["Many To Many variable"](https://docs.sqlalchemy.org/en/13/orm/basic_relationships.html#many-to-many)
-  [Association Object](https://docs.sqlalchemy.org/en/13/orm/basic_relationships.html#association-object) 

Where `'association'` is name of relation table, for/from example `UserGroup`

File: app/models.py (hunk 1 above)
Comment (filipjurcak) [reply app/models.py]: Also, email seems to me equally important as telephone number (in terms of privacy). So we should either remove both from the `__repr__` or we include them both. 

What is your opinion? (personally, I would stick with having them both in `__repr__`)

**Comment by filipjurcak** [comment]
Okay, I updated the PR with updated column names for password and salt, are we okay to merge? @matusf @Saamoot

**Comment by Saamoot** [comment]
I don't see any immediate visible issues, so yes.

**Comment by Saamoot** [comment]
I don't see any immediate visible issues, so yes.

**Review by matusf — COMMENTED** [review]
lgtm

**Pull request merged**